### ISSUE TITLE
"Project requires Cabal but it isn't installed" but cabal is installed and on the PATH (linux)

### ISSUE THREAD

**Issue by avh4** [haskell/vscode-haskell#303]
When opening a .hs file, an error is shown: "Project requires Cabal but it isn't installed / Source: Haskell (Extension)"  Clicking the button in the error opens https://www.haskell.org/ghcup/ in my web browser.  But cabal is already installed with ghcup.  Opening a terminal in VSCode and running `which cabal` gives `/home/avh4/.ghcup/bin/cabal`

Haskell extension v1.1.0
platform: Ubuntu Linux 20.04.1

**Comment by jneira** [comment]
Maybe there is some mismatch in setting the `PATH` between `.profile` and `bashrc`, is the extension using the former?
//cc @bubba

**Comment by bubba** [comment]
@avh4 if you open a terminal window (not in vscode) and then launch vs code by `code .`, does it work? When launching vscode from an application launcher it might not have the same PATH

**Comment by avh4** [comment]
yes, looks like launching `code` from the command line make it work.

Is there a way to set the path or cabal location more robustly so that it will work regardless of how the editor is launched?

**Comment by bubba** [comment]
@avh4  not at the moment no, out of curiosity what desktop environment are you using? It might be useful to stick some troubleshooting information in the README

**Comment by CristianSimion5** [comment]
Got the same issue, except the error says that ghc is not installed instead of cabal. Launching vscode from the command line makes it work. I explored the application launcher shortcut and it uses the `--no-sandbox --unity-launch %F` arguments. `--no-sandbox` seems to be the cause in my case. I am using Manjaro KDE 20.1.

Can someone explain to me what happens to the `PATH` when using no sandbox?

**Comment by js-musedev** [comment]
I have a similar issue where the Haskell extension somehow uses an older version of `cabal` than the one returned by `which cabal` (and for which I can't see how it could be on my path at all). Starting `code .` from my project folder sort of fixes that, but surfaces more build issues that don't come up when I run `cabal build all` from the shell.

**Comment by bubba** [comment]
I think we might be able to mitigate a lot of these issues if we execute these commands via the shell I.e `sh -c 'cabal whatever'`
